src/neo4j_genai/pipeline/component.py

```diff
@@ -0,0 +1,79 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from __future__ import annotations
+
+import abc
+import inspect
+from typing import Any, get_type_hints
+
+from pydantic import BaseModel
+
+
+class DataModel(BaseModel):
+    """Input or Output data model for Components"""
+
+    pass
+
+
+class ComponentMeta(abc.ABCMeta):
+    def __new__(
+        meta, name: str, bases: tuple[type, ...], attrs: dict[str, Any]
+    ) -> type:
+        # extract required inputs from the run method signature
+        run_method = attrs.get("run")
+        if run_method is not None:
+            sig = inspect.signature(run_method)
+            attrs["component_inputs"] = {
+                param.name: {
+                    "has_default": param.default != inspect.Parameter.empty,
+                    "annotation": param.annotation,
+                }
+                for param in sig.parameters.values()
+                if param.name not in ("self",)
+            }
+        # extract returned fields from the run method return type hint
+        return_model = get_type_hints(run_method).get("return")
+        # the type hint must be a subclass of DataModel
+        # but not DataModel itself which is empty
+        if name != "Component":
```

Comment:
What would happen if name == "Component"?

Reply:
It would raise an error because the annotation of the run method is `DataModel`. But actually I think we can get rid of this test, some components might not return anything and keep using the `DataModel` class.